### DISCUSSION TITLE
Is it possible to display a modal based on folders that exist?

### DISCUSSION THREAD

**Discussion by daamiian** [SilentVoid13/Templater#1116] [Help] [answered]
I have a `projects ` folder and a `courses` folder, each containing folders of each project / course.

I use a project/course ID prefix, so a project dashboard might have the title: `ENG-DA English with Damian`. A project doc specifically relevant to the project might read: `ENG-DA Learning the First 100 Words in English`. This way at a glance while reading links, I can tell what project/course it was from. 

While creating docs, my current workflow moves the doc to the `projects` or `courses` folder, then I manually move it to the corresponding project / course folder (`ENG-DA` folder for example). It's a step I feel could be automated. 

I'd love it if I could create a selector modal that displays all folders in the projects folder, and allow me to pick from it, then it moves the file to that location. But I'm not sure if that's even possible or how to go about it.

Any advice is appreciated, thank you

**Comment by Zachatoo** [comment, marked as answer]
Sure can! Here's one way to do that:

```js
<%*
// Get details about folder (can change to whatever folder you want)
const folder = "projects";
const tFolder = app.vault.getAbstractFileByPath(folder);

// Some checks to make sure it's actually a folder and that it has any subfolders
if (!("children" in tFolder)) {
  throw new Error("Folder is not a folder.");
}
const subfolders = tFolder.children.filter(subfolder => "children" in subfolder);
if (subfolders.length === 0) {
  throw new Error("Folder doesn't have any subfolders.");
}

// Prompt user to select subfolder
const selectedFolder = (await tp.system.suggester((subfolder) => subfolder.name, subfolders)).path;

// Move file to subfolder
await tp.file.move(`${selectedFolder}/${tp.file.title}`);
-%>
```

Relevant links: https://zachyoung.dev/snippets/templater#suggester-for-subfolders-in-a-specific-folder

**Reply by daamiian** [reply]
Maybe I'm missing something, but I put this snippet inside my course doc template, and now when I create a new course doc, I get an error. I don't use templater for much more than moving files, suggestor, and prompt. I put these in the template I need and usually it works. 

Just to make my workflow a bit clearer (I'm open to suggestions as well), I have a course doc template, and a project doc template. I have this at the bottom of my course template:
`<% await tp.file.move("/Courses/" + tp.file.title) %>`
and it moves the course doc to the general courses folder. I don't use script files because I don't know what I'm doing 😂

Edit: I realized I made a mistake by putting "courses" instead of "Courses", my real folder name. But now it works as intended, thank you for the help!
